core/core_integrations/slack.js

```diff
@@ -274,6 +270,35 @@ var request = require.safe('request'),
         });
     },
 
+    connect = function(connectionDetails) {
+        var socket;
+        console.debug('slack-> Connecting to socket');
+
+        if (connectionDetails) {
+            socket = new WebSocket(connectionDetails.url);
+            console.debug(connectionDetails.url);
+            sockets[connectionDetails.team_id] = socket;
+            socket.on('open', function() {
+                console.debug('slack-> Connection to team: ' + teamData[connectionDetails.team_id].team.name + ' established');
+            }).on('message', function(data) {
+                startPingPongTimer(connectionDetails.team_id);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["startPingPongTimer" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2651361702/issues/source?bid=3282987&fileBranchId=3324570#l284)